src/xz/file_io.c

```diff
@@ -1158,13 +1186,20 @@ extern size_t
 io_read(file_pair *pair, io_buf *buf, size_t size)
 {
 	// We use small buffers here.
-	assert(size < SSIZE_MAX);
+	assert(size < (size_t)SSIZE_MAX);
```

Comment:
I changed these assertions to use `IO_BUFFER_SIZE`. Then the changes to `io_read` and `io_write_buf` to limit to `UINT32_MAX` shouldn't be needed either.